server/global/data_analyzer/analyzer.js

```diff
@@ -0,0 +1,296 @@
+
+  warnings.push(...analyzeMolecules(data.molecules));
+
+  return warnings;
+}
+
+/**
+ * Analyze a property
+ * @param {string} property The property name
+ * @param {{id : number, name : string}[]} values
+ * @returns {AnalyzerWarning[]}
+ */
+function analyzeProperty(property, values) {
+  const names = values.map((v) => v.name);
+  const tooLongValues = getTooLongValues(names, MAX_LENGTH.PROPERTY_VALUE);
+  const closeValues = getTooCloseValues(names, PROPERTY_VALUE_MIN_DISTANCE);
+  const noStringValue = names.filter((name) => !isString(name));
+
+  return [
+    ...tooLongValues.map(
+      (value) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.TOO_LONG_VALUE,
+          `La valeur "${value}" de la propriété "${property}" est trop longue (max ${MAX_LENGTH.PROPERTY_VALUE})`
+        )
+    ),
+    ...closeValues.map(
+      (group) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.TOO_CLOSE_VALUES,
+          `Ces valeurs de "${property}" sont très proches : "${group.join('", "')}"`
+        )
+    ),
+    ...noStringValue.map(
+      (value) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.INVALID_TYPE,
+          `Une valeur de "${property}" devrait être une chaine de caractères : "${value}"`
+        )
+    ),
+  ];
+}
+
+/**
+ * Analyze a classification
+ * @param {string} classification The classification name
+ * @param {object[]} nodes The classification node
+ * @returns {AnalyzerWarning[]}
+ */
+function analyzeClassification(classification, nodes) {
+  const names = flattenClassification(nodes).map((n) => n.name);
+  const closeValues = getTooCloseValues(names, CLASSIFICATION_VALUE_MIN_DISTANCE);
+  const tooLongValues = getTooLongValues(names, MAX_LENGTH.CLASSIFICATION_VALUE);
+  const nodesHavingSeveralParents = findNodeWithDifferentsParents(nodes);
+  const noStringValue = names.filter((name) => !isString(name));
+
+  return [
+    ...closeValues.map(
+      (group) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.TOO_CLOSE_VALUES,
+          `Ces valeurs de "${classification}" sont très proches : "${group.join('", "')}"`
+        )
+    ),
+    ...nodesHavingSeveralParents.map(
+      (node) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.DUPLICATE_CLASSIFICATION_NODE,
+          `La valeur de "${classification}" "${
+            node.node
+          }" apparait plusieurs fois dans la hiérarchie, enfant de : "${node.parents.join('", "')}"`
+        )
+    ),
+    ...tooLongValues.map(
+      (value) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.TOO_LONG_VALUE,
+          `La valeur de "${classification}" "${value}" est trop longue (max ${MAX_LENGTH.CLASSIFICATION_VALUE})`
+        )
+    ),
+    ...noStringValue.map(
+      (value) =>
+        new AnalyzerWarning(
+          AnalyzerWarning.INVALID_TYPE,
+          `Une valeur de '${classification}' devrait être une chaîne de caractères : '${value}'`
```

Comment:
Double quotes could be great here too :)

Reply:
oups !